stl/src/xonce.cpp

```diff
@@ -17,13 +23,15 @@ _CRTIMP2_PURE int __CLRCALL_PURE_OR_CDECL _Execute_once(
     // we introduce _Xfg_trampoline which has PINIT_ONCE_FN's type signature and
     // calls into _Callback as an _Execute_once_fp_t for XFG compatibility.
 
+    _Xfg_trampoline_parameter _Trampoline_parameter = {_Pv, _Callback};
+
     PINIT_ONCE_FN _Xfg_trampoline = [](PINIT_ONCE _InitOnce, PVOID _Parameter, PVOID* _Context) {
-        const auto _Callback = reinterpret_cast<_Execute_once_fp_t>(_Context);
-        return static_cast<BOOL>(_Callback(_InitOnce, _Parameter, nullptr));
+        const auto _Trampoline_parameter = static_cast<_Xfg_trampoline_parameter*>(_Parameter);
+        return static_cast<BOOL>(_Trampoline_parameter->_Callback(_InitOnce, _Trampoline_parameter->_Pv, _Context));
     };
 
-    return InitOnceExecuteOnce(
-        reinterpret_cast<PINIT_ONCE>(&_Flag._Opaque), _Xfg_trampoline, _Pv, reinterpret_cast<PVOID*>(_Callback));
+    return InitOnceExecuteOnce(reinterpret_cast<PINIT_ONCE>(&_Flag._Opaque), _Xfg_trampoline,
+        static_cast<PVOID>(&_Trampoline_parameter), nullptr);

```

Comment:
I observe that this `static_cast<PVOID>` is unnecessary (it will implicitly convert) but I'm ok with checking this in as-is since I didn't notice it before.